doc/kafka_rmq.md

```diff
@@ -0,0 +1,183 @@
+                                                     :non-recoverable-exception {:enabled true
+                                                                                 :back-off-ms 1000
+                                                                                 :count 5}}}}}}
+```
+
+- `:hosts` is a comma separated values of RabbitMQ hostnames (dns-names OR IPs).
+- `:port` specifies the port number on which the RabbitMQ nodes are running.
+- `:channel-pool` specifies the properties for the RabbitMQ channel pool used for publishing
+- `:address-resolver` specifies the strategy to figure out RabbitMQ hosts IP addresses. `:dns` is the default and shoud
+  be used when `:hosts` specifies a DNS address. `:ip-list` should be used when comma separated IPs are provided.
+- `:publish-retry` defines the config for recoverable and non-recoverable exceptions.
+    - Recoverable exceptions
+        - `:back-off-ms` - defines the time period after which a retry should happen
+    - Non-recoverable exceptions
+        - `:enabled` - defines whether retries should happen
+        - `:back-off-ms` - defines the time period after which a retry should happen
+        - `:count` - defines the number of retries
+- By default, your queues and exchanges are replicated across (n+1)/2 nodes in the cluster
+
+## Exponential Backoff based Retries
+
+In addition to linear retries, Ziggurat users can now use exponential backoff strategy for retries. This means that the message
+timeouts after every retry increase by a factor of 2. So, if your configured timeout is 100ms the backoffs will have timeouts as
+`200, 300, 700, 1500 ..`. These timeouts are calculated using the formula `(queue-timeout-ms * ((2**exponent) - 1))` where `exponent` falls in this range `[1,(min 25, configured-retry-count)]`.
+
+The number of retries possible in this case are capped at 25.
+
+The number of queues created in the RabbitMQ are equal to the configured-retry-count or 25, whichever is smaller.
+
+Exponential retries can be configured as described below.
+
+```$xslt
+:ziggurat {:stream-router {:default {:application-id "application_name"...}}}
+           :retry         {:type   [:exponential :keyword]
+                           :count  [10 :int]
+                           :enable [true :bool]}
+
+```
+
+Exponential retries can be configured for channels too. Additionally, a user can specify a custom `queue-timeout-ms` value per channel.
+Timeouts for exponential backoffs are calculated using `queue-timeout-ms`. This implies that each channel can have separate count of retries
+and different timeout values.
+
+```$xslt
+:ziggurat {:stream-router {:default {:application-id "application_name"...
+                                     :channels {:channel-1 .....
+                                                           :retry {:type   [:exponential :keyword]
+                                                                                      :count  [10 :int]
```

Comment:
Also explain concept of RMQ worker count config and prefetch relationship

Reply:
metioned the relationship between the two below this.